File: 48-RotateImage/48-RotateImage.cpp
// Last updated: 6/11/2026, 11:37:48 AM
class Solution {
public:
    int n;
    int m;
    void rotate(vector<vector<int>>& arr) {
        n = arr.size();
        m = arr[0].size();
        for(int i=0; i<n; i++){
            for(int j=0; j<i; j++){
                swap(arr[i][j], arr[j][i]);
            }
        }

        for(int i=0; i<n; i++){
            int j1 = 0;
            int j2 = m-1;
            while(j1<j2){
                swap(arr[i][j1], arr[i][j2]);
                j1++;
                j2--;
            }
        }

        // return arr;
    }
};

File: 48-RotateImage/48-RotateImage.py
# Last updated: 6/11/2026, 11:37:54 AM
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if(i!=j and i<=j):
                    matrix[i][j],matrix[j][i] = matrix[j][i],matrix[i][j]

        print(matrix)
        for j in range(len(matrix[0])//2):
            for i in range(len(matrix)):
                matrix[i][j],matrix[i][len(matrix[0])-j-1] = matrix[i][len(matrix[0])-j-1],matrix[i][j]

        